server/chat-plugins/chatlog.ts

```diff
@@ -353,24 +591,39 @@ export const pages: PageTable = {
 		}
 
 		void accessLog.writeLine(`${user.id}: <${roomid}> ${date}`);
-
 		this.title = '[Logs] ' + roomid;
 
-		if (date) {
+		const hasSearch = opts?.includes('search-') || opts?.includes('csearch-');
+		const context = opts?.includes('csearch-');
+		const search = opts?.slice(7);
+		const isAll = (toID(date) === 'all' || toID(date) === 'alltime');
+
+		const parsedDate = new Date(date as string);
+		// this is apparently the best way to tell if a date is invalid
+		if (isNaN(parsedDate.getTime()) && !isAll && date !== 'today') {
+			return LogViewer.error(`Invalid date.`);
+		}
+
+		if (date && !hasSearch) {
 			if (date === 'today') {
 				return LogViewer.day(roomid, LogReader.today(), opts);
-			}
-			const parsedDate = new Date(date);
-			// this is apparently the best way to tell if a date is invalid
-			if (isNaN(parsedDate.getTime())) return LogViewer.error(`Invalid date.`);
-
-			if (date.split('-').length === 3) {
+			} else if (date.split('-').length === 3) {
 				return LogViewer.day(roomid, parsedDate.toISOString().slice(0, 10), opts);
 			} else {
 				return LogViewer.month(roomid, parsedDate.toISOString().slice(0, 7));
 			}
+		} else if (date && hasSearch && search) {
+			if (context) {
```

Comment:
Ripgrep does support context..

What I'd do is: Add a `Config.chatlogsreader` where it can be `fs` or `ripgrep`, and then run `fsSearch` or `ripgrepSearch` corresponding to the config option.